src/utils/array.ts

```diff
@@ -14,3 +14,24 @@ E.g. interpose([1,2,3], '+') -> [1, '+', 2, '+', 3]
 export function interpose<T, U>(array: Array<T>, delim: U) {
   return flatMap(array, (el) => [delim, el]).slice(1);
 }
+
+/*
+Takes an array and returns an array of pairs: {current, next}
+E.g. iterateWithNext([1, 2, 3, 4]) -> [{1, 2}, {2, 3}, {3, 4}]
+Warning: when an array of 1 element is given, no pairs are returned []
+ */
+export function iterateWithNext<T>(
+  iterable: Array<T>,
+): {current: T; next: T}[] {
+  function* generator<T>(iterable: Array<T>) {
```

Comment:
Gøy med generator! Når jeg ser dette så begynte jeg å tenke på om generator-funksjon hadde vært en lur løsning på reisesøk, hvor man kunne kalt generatoren for hver paginering, også hadde man yieldet ut nye resultater fra Entur.